--- module-init-tools-3.4/insmod.c.orig	2007-10-07 23:51:46.000000000 +0200
+++ module-init-tools-3.4/insmod.c	2008-08-24 00:27:48.322128443 +0200
@@ -28,6 +28,7 @@
 #include <errno.h>
 #include <asm/unistd.h>
 
+#include "zlibsupport.h"
 #include "backwards_compat.c"
 
 #define streq(a,b) (strcmp((a),(b)) == 0)
@@ -57,48 +58,6 @@
 	}
 }
 
-static void *grab_file(const char *filename, unsigned long *size)
-{
-	unsigned int max = 16384;
-	int ret, fd, err_save;
-	void *buffer = malloc(max);
-	if (!buffer)
-		return NULL;
-
-	if (streq(filename, "-"))
-		fd = dup(STDIN_FILENO);
-	else
-		fd = open(filename, O_RDONLY, 0);
-
-	if (fd < 0)
-		return NULL;
-
-	*size = 0;
-	while ((ret = read(fd, buffer + *size, max - *size)) > 0) {
-		*size += ret;
-		if (*size == max) {
-			void *p;
-
-			p = realloc(buffer, max *= 2);
-			if (!p)
-				goto out_error;
-			buffer = p;
-		}
-	}
-	if (ret < 0)
-		goto out_error;
-
-	close(fd);
-	return buffer;
-
-out_error:
-	err_save = errno;
-	free(buffer);
-	close(fd);
-	errno = err_save;
-	return NULL;
-}
-
 int main(int argc, char *argv[])
 {
 	unsigned int i;
diff -urN module-init-tools-3.0.org/Makefile.am module-init-tools-3.0/Makefile.am
--- module-init-tools-3.0.org/Makefile.am	2004-07-17 19:56:16.090395360 +0200
+++ module-init-tools-3.0/Makefile.am	2004-07-17 20:05:45.953762960 +0200
@@ -1,4 +1,4 @@
-insmod_SOURCES = insmod.c testing.h
+insmod_SOURCES = insmod.c testing.h zlibsupport.c zlibsupport.h
 lsmod_SOURCES = lsmod.c testing.h
 modprobe_SOURCES = modprobe.c logging.c index.c zlibsupport.c logging.h index.h testing.h zlibsupport.h
 rmmod_SOURCES = rmmod.c testing.h
@@ -6,10 +6,8 @@
 modinfo_SOURCES = modinfo.c zlibsupport.c testing.h zlibsupport.h
 modindex_SOURCES = modindex.c logging.c index.c logging.h index.h
 
-insmod_static_SOURCES = insmod.c
+insmod_static_SOURCES = insmod.c zlibsupport.c zlibsupport.h
 insmod_static_LDFLAGS = -static
-# We don't want the $(zlib_flags) here: that makes a dynamic executable
-insmod_static_LDADD = 
 
 modprobe_LDADD = $(ZLIB)
 depmod_LDADD = $(ZLIB)
