--- module-init-tools-3.12/modprobe.c.orig	2010-05-04 07:19:27.000000000 +0200
+++ module-init-tools-3.12/modprobe.c	2010-08-15 09:15:37.990822457 +0200
@@ -82,6 +82,8 @@
 
 } modprobe_flags_t;
 
+char *modprobe_d_ver = NULL;
+
 #ifndef MODULE_DIR
 #define MODULE_DIR "/lib/modules"
 #endif
@@ -1065,19 +1067,27 @@
 		/* sort files from directory into list */
 		while ((i = readdir(dir)) != NULL) {
 			size_t len;
+			struct stat statbuf;
+			char statpath[PATH_MAX];
 
 			if (i->d_name[0] == '.')
 				continue;
 			if (!config_filter(i->d_name))
 				continue;
 
+			snprintf(statpath, sizeof(statpath), "%s/%s", filename, i->d_name);
+			if ((stat(statpath, &statbuf) == 0) && (S_ISDIR(statbuf.st_mode)))
+				continue;
+
 			len = strlen(i->d_name);
 			if (len < 6 ||
 			    (strcmp(&i->d_name[len-5], ".conf") != 0 &&
-			     strcmp(&i->d_name[len-6], ".alias") != 0))
+			     strcmp(&i->d_name[len-6], ".alias") != 0)) {
 				warn("All config files need .conf: %s/%s, "
-				     "it will be ignored in a future release.\n",
+				     "Ignoring non conformant file.\n",
 				     filename, i->d_name);
+				continue;
+			}
 			fe = malloc(sizeof(struct file_entry) + len + 1);
 			if (fe == NULL)
 				continue;
@@ -1129,6 +1139,10 @@
 		warn("Deprecated config file /etc/modprobe.conf, "
 		      "all config files belong into /etc/modprobe.d/.\n");
 
+	/* kernel ver based config */
+	if ( modprobe_d_ver != NULL )
+			parse_config_scan(modprobe_d_ver, conf, dump_only, removing);
+
 	/* default config */
 	parse_config_scan("/etc/modprobe.d", conf, dump_only, removing);
 }
@@ -1815,6 +1829,7 @@
 		print_usage(argv[0]);
 
 	nofail_asprintf(&dirname, "%s%s/%s", basedir, MODULE_DIR, buf.release);
+	nofail_asprintf(&modprobe_d_ver, "/etc/modprobe.d/%s", buf.release);
 
 	/* Old-style -t xxx wildcard?  Only with -l. */
 	if (list_only) {
