--- module-init-tools-3.3-pre11/doc/depmod.sgml	2007-03-22 09:36:02.000000000 +0200
+++ module-init-tools-3.3-pre11-docs/doc/depmod.sgml	2007-12-15 18:29:03.003914335 +0200
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+[
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- module-init-tools-3.3-pre11/doc/insmod.sgml	2007-03-22 09:36:02.000000000 +0200
+++ module-init-tools-3.3-pre11-docs/doc/insmod.sgml	2007-12-15 18:29:28.165359303 +0200
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+[
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- module-init-tools-3.3-pre11/doc/lsmod.sgml	2007-03-22 09:36:02.000000000 +0200
+++ module-init-tools-3.3-pre11-docs/doc/lsmod.sgml	2007-12-15 18:30:16.680145031 +0200
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+[
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- module-init-tools-3.3-pre11/doc/modinfo.sgml	2007-03-22 09:36:02.000000000 +0200
+++ module-init-tools-3.3-pre11-docs/doc/modinfo.sgml	2007-12-15 18:31:24.452036477 +0200
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+[
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- module-init-tools-3.3-pre11/doc/modprobe.conf.sgml	2007-03-22 09:36:02.000000000 +0200
+++ module-init-tools-3.3-pre11-docs/doc/modprobe.conf.sgml	2007-12-15 18:26:56.712661367 +0200
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+[
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- module-init-tools-3.3-pre11/doc/modprobe.sgml	2007-03-22 09:36:02.000000000 +0200
+++ module-init-tools-3.3-pre11-docs/doc/modprobe.sgml	2007-12-15 18:31:06.923029971 +0200
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+[
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- module-init-tools-3.3-pre11/doc/rmmod.sgml	2007-03-22 09:36:02.000000000 +0200
+++ module-init-tools-3.3-pre11-docs/doc/rmmod.sgml	2007-12-15 18:30:38.433394102 +0200
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+[
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
