--- module-init-tools-0.9.9/Makefile.am.wiget	Fri Jan 17 08:47:19 2003
+++ module-init-tools-0.9.9/Makefile.am	Tue Feb 11 15:26:48 2003
@@ -38,8 +38,8 @@
 mandir = $(shell if [ `echo $(prefix)/ | tr -s /` = / ]; then echo /usr/share/man; else echo $(prefix)/man; fi)
 
 installmanpages:
-	for f in $(SGML8:doc/%.sgml=%.8); do $(INSTALL) -m 644 $(srcdir)/$$f $(mandir)/man8; done
-	for f in $(SGML5:doc/%.sgml=%.5); do $(INSTALL) -m 644 $(srcdir)/$$f $(mandir)/man5; done
+	for f in $(SGML8:doc/%.sgml=%.8); do $(INSTALL) -m 644 $(srcdir)/$$f $(DESTDIR)$(mandir)/man8; done
+	for f in $(SGML5:doc/%.sgml=%.5); do $(INSTALL) -m 644 $(srcdir)/$$f $(DESTDIR)$(mandir)/man5; done
 
 distclean: killmanpages
 
