diff -Nur module-init-tools-3.2.2/modprobe.c module-init-tools-3.2.2.modprobe_d/modprobe.c
--- module-init-tools-3.2.2/modprobe.c	2005-12-02 00:42:09.000000000 +0100
+++ module-init-tools-3.2.2.modprobe_d/modprobe.c	2006-01-21 03:11:18.000000000 +0100
@@ -73,6 +73,8 @@
 
 } modprobe_flags_t;
 
+char *modprobe_d_ver = NULL;
+
 #ifndef MODULE_DIR
 #define MODULE_DIR "/lib/modules"
 #endif
@@ -876,10 +878,12 @@
 			len = strlen(i->d_name);
 			if (len < 6 ||
 			    (strcmp(&i->d_name[len-5], ".conf") != 0 &&
-			     strcmp(&i->d_name[len-6], ".alias") != 0))
+			     strcmp(&i->d_name[len-6], ".alias") != 0)) {
 				warn("All config files need .conf: %s/%s, "
-				     "it will be ignored in a future release.\n",
+				     "Ignoring non conformant file.\n",
 				     filename, i->d_name);
+				continue;
+			}
 			fe = malloc(sizeof(struct file_entry) + len + 1);
 			if (fe == NULL)
 				continue;
@@ -940,6 +944,11 @@
 		warn("Deprecated config file /etc/modprobe.conf, "
 		      "all config files belong into /etc/modprobe.d/.\n");
 
+	/* kernel ver based config */
+	if ( modprobe_d_ver != NULL )
+			parse_config_scan(modprobe_d_ver, name, dump_only, removing,
+							options, commands, aliases, blacklist);
+
 	/* default config */
 	parse_config_scan("/etc/modprobe.d", name, dump_only, removing,
 			  options, commands, aliases, blacklist);
@@ -1509,6 +1518,7 @@
 		print_usage(argv[0]);
 
 	nofail_asprintf(&dirname, "%s%s/%s", basedir, MODULE_DIR, buf.release);
+	nofail_asprintf(&modprobe_d_ver, "/etc/modprobe.d/%s", buf.release);
 	nofail_asprintf(&aliasfilename, "%s/modules.alias", dirname);
 	nofail_asprintf(&symfilename, "%s/modules.symbols", dirname);
 
