--- ./modprobe.c.org	Fri May  9 10:28:32 2003
+++ ./modprobe.c	Sun Jun 15 14:08:15 2003
@@ -998,8 +998,11 @@
 
 			if (!wildcard || !realname)
 				grammar(cmd, filename, linenum);
-			else if (fnmatch(wildcard,name,0) == 0)
-				result = NOFAIL(strdup(realname));
+			else if (fnmatch(wildcard,name,0) == 0) {
+				if (result)
+					free(result);
+				name = result = NOFAIL(strdup(realname));
+			}
 		} else if (strcmp(cmd, "include") == 0) {
 			char *newresult, *newfilename;
 
