--- module-init-tools-3.0-pre9/configure.ac.orig	2004-01-30 05:15:25.000000000 +0100
+++ module-init-tools-3.0-pre9/configure.ac	2004-01-30 23:08:59.729518344 +0100
@@ -14,7 +14,7 @@
 [  --enable-zlib               Handle gzipped modules],
 [if test "$enableval" = "yes"; then
   AC_DEFINE(CONFIG_USE_ZLIB)
-  zlib_flags="-Wl,-Bstatic -lz -Wl,-Bdynamic"
+  zlib_flags="-lz"
 fi])
 
 AC_PROG_CC
@@ -27,8 +27,8 @@
  
 # Delay adding the zlib_flags until after AC_PROG_CC, so we can distinguish
 # between a broken cc and a working cc but missing libz.a.
-LDADD="$LDADD $zlib_flags"
-AC_SUBST(LDADD)
+ZLIB="$zlib_flags"
+AC_SUBST(ZLIB)
 
 case $target in
 *-*-linux*) ;;
--- module-init-tools-0.9.14/Makefile.am.orig	Tue Aug 26 04:13:18 2003
+++ module-init-tools-0.9.14/Makefile.am	Mon Oct 13 10:59:14 2003
@@ -13,6 +13,12 @@
 # We don't want the $(zlib_flags) here: that makes a dynamic executable
 insmod_static_LDADD = 
 
+modprobe_LDADD = $(ZLIB)
+depmod_LDADD = $(ZLIB)
+modinfo_LDADD = $(ZLIB)
+insmod_LDADD = $(ZLIB)
+insmod_static_LDADD = $(ZLIB)
+
 EXTRA_insmod_SOURCES =
 EXTRA_lsmod_SOURCES =
 EXTRA_modprobe_SOURCES = 
